src/components/MultiHopTrade/components/TradeInput/TradeInput.tsx

```diff
@@ -162,6 +162,8 @@ export const TradeInput = memo(() => {
     // WARNING: do not remove.
     // clear the confirmed quote on mount to prevent stale data affecting the selectors
     dispatch(tradeQuoteSlice.actions.resetConfirmedQuote())
+    // clear the active quote index on mount to prevent stale data affecting the selectors
+    dispatch(tradeQuoteSlice.actions.resetActiveQuoteIndex())
```

Comment:
FYI this commit @woodenfurniture @0xApotheosis

We don't opt chain to 0 on `selectActiveQuoteIndex` anymore, that was semantically wrong, and we were getting stale quotes all over the place when trying to rely on `selectActiveQuote`, because even if we clear the index  on mount, we would still opt chain to 0.

This means we now:

- always return `undefined` as is from `selectActiveQuoteIndex` and opt chain to 0 when actually needed
- dispatch `setActiveQuoteIndex(0`  on quotes received so the behavior is effectively the same as before (with the added UX bonus of not having a "ghost selected quote" when quotes are fetching)
- handle this in `selectActiveSwapperApiResponse` and return `undefined` in case of undefined active quote index. This is the guts of this commit in ensuring we never have stale quotes

Reply:
Confirmed with apo juice we're fine with this, let's get this in